### PR TITLE
Add documentation for SocketReceiveMultiMsgApi extension.

### PR DESCRIPTION
**Pull request by jellefoks**
This adds documentation for platforms to integrate CobaltExtensionSocketReceiveMultiMsgApi

b/376351154
b/205134049
b/376350096

### CODE REVIEW

File: starboard/doc/networking_performance.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+     "//starboard/shared/posix/socket_resolve.cc",
+     "//starboard/shared/posix/socket_send_to.cc",
+     "//starboard/shared/posix/socket_set_broadcast.cc",
+```
+
+2.  In the platform's `system_get_extensions.cc`: 
+    Add two includes and a three line if statement.
+
+```
+diff --git a/starboard/linux/shared/system_get_extensions.cc b/starboard/linux/shared/system_get_extensions.cc
```
Comment (kaidokert) [inline starboard/doc/networking_performance.md:50]: I wouldn't copy code in here - just a perma-link to the implementation is fine ( can use a tag link or such to ensure it stays around )

File: starboard/doc/networking_performance.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
+It is strongly encouraged for platforms to include the
```
Comment (kaidokert) [inline starboard/doc/networking_performance.md:1]: Please add a bit of a context-setting header here. E.g. `## Overview`, Cobalt 25 implements IETF HTTP/3 and benefits from more specific tuning and customizations to improve its performance and such.

File: starboard/doc/networking_performance.md (hunk 1 above)
Comment (jellefoks) [reply starboard/doc/networking_performance.md:50]: Github gives no option to perma-link to the diff of only a single file, and doesn't allow a link with multiple lines that are not together. The PR also includes the implementation, but using it only takes edits to these two files.

The files themselves already have many other lines of code. A developer looking to implement would greatly benefit from a diff versus the entire implementaton PR or the entire files.

File: starboard/doc/networking_performance.md (hunk 2 above)
Comment (jellefoks) [reply starboard/doc/networking_performance.md:11]: Done.

File: starboard/doc/networking_performance.md (hunk 1 above)
Comment (jellefoks) [reply starboard/doc/networking_performance.md]: Changed to a patch command, with shorter context.

**Review by y4vor — APPROVED** [review]

**Pull request merged**